catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -522,6 +531,127 @@ public void init() {
+                    .toString();
+            String user = getSubjectEmail();
+            if (user == null) {
+                return util.getResponseWrapper("error",
+                        "You are not authorized to create notes! A user email is required. "
+                                + "Please ensure you are logged in and/or have a valid email registered in the system.");
+            }
+            if (StringUtils.isBlank(annotation)) {
+                return util.getResponseWrapper("error", "No annotation!");
+            }
+            NoteMetacard noteMetacard = new NoteMetacard(queryId, user, annotation);
+
+            Metacard workspaceMetacard = util.findWorkspace(workspaceId);
+
+            if (workspaceMetacard == null) {
+                return util.getResponseWrapper("error", "Cannot find the workspace metacard!");
+            }
+
+            util.copyAttributes(workspaceMetacard, SECURITY_ATTRIBUTES, noteMetacard);
+
+            Metacard note = saveMetacard(noteMetacard);
+
+            SecurityLogger.auditWarn(String.format(
+                    "Attaching an annotation to a resource: resource=%s annotation=%s",
+                    workspaceId,
+                    noteMetacard.getId()), SecurityUtils.getSubject());
+
+            Map<String, String> responseNote = NoteUtil.getResponseNote(note);
+            if (responseNote == null) {
+                throw new IllegalStateException("An error occurred while storing the annotation");
+            }
+            return util.getResponseWrapper("success", util.getJson(responseNote));
+        });
+
+        get("/annotations/:queryid", (req, res) -> {
+            String queryId = req.params(":queryid");
+
+            List<Metacard> retrievedMetacards =
+                    util.getAssociatedMetacardsByTwoAttributes(Note.PARENT_ID,
+                            Core.METACARD_TAGS,
+                            queryId,
+                            "note");
+            if (!retrievedMetacards.isEmpty()) {
+                ArrayList<String> getResponse = new ArrayList<>();
+                retrievedMetacards.sort(Comparator.comparing(Metacard::getCreatedDate));
+                for (Metacard metacard : retrievedMetacards) {
+                    Map<String, String> responseNote = NoteUtil.getResponseNote(metacard);
+                    if (responseNote != null) {
+                        getResponse.add(util.getJson(responseNote));
+                    }
+                }
+                return util.getResponseWrapper("success", getResponse.toString());
+            } else {
+                return util.getResponseWrapper("error",
+                        "Could not find any annotations for this result!");
+            }
+        });
+
+        put("/annotations/:id", APPLICATION_JSON, (req, res) -> {
+            Map<String, Object> incoming = JsonFactory.create()
+                    .parser()
+                    .parseMap(req.body());
```

Comment:
❗️ This needs to be `util.safeGetBody(req)`

Reply:
fixed